### PR TITLE
add UserNameChecked | delete EmailCode | fix display UserName

### PR DESCRIPTION
**Pull request by Ketteiteki**
Fixes https://github.com/MangoInstantMessenger/MangoMessengerAPI/issues/378

### CODE REVIEW

File: MangoAPI.Client/src/app/components/settings/settings.component.ts [hunk 1]
```diff
@@ -78,6 +79,12 @@ export class SettingsComponent implements OnInit, OnDestroy {
       .subscribe({
         next: (response) => {
           this.currentUser = response.user;
+          
+          console.log(response.user);
+          if (response.user.userNameChanged === false) {
+            console.log(2);
```
Comment (kolosovpetro) [inline MangoAPI.Client/src/app/components/settings/settings.component.ts:85]: remove this console log

File: MangoAPI.BusinessLogic/ApiCommands/Users/UpdateUserAccountInfoCommandHandler.cs [hunk 2]
```diff
@@ -62,6 +62,19 @@ public class
             dbContext.Chats.UpdateRange(userChats);
         }
 
+        if (user.UserName != request.Username)
+        {
+            user.UserNameChanged = true;
+
+            if (Guid.TryParse(request.Username, out _))
```
Comment (kolosovpetro) [inline MangoAPI.BusinessLogic/ApiCommands/Users/UpdateUserAccountInfoCommandHandler.cs:69]: Move this validation to the `UpdateUserAccountInfoCommandValidator.cs`

File: MangoAPI.Domain/Entities/UserEntity.cs [hunk 3]
```diff
@@ -13,8 +13,8 @@ public sealed class UserEntity : IdentityUser<Guid>
 
     public string Bio { get; set; }
 
-    public Guid EmailCode { get; set; }
```
Comment (kolosovpetro) [inline MangoAPI.Domain/Entities/UserEntity.cs:16]: Have you updated migrations after email code removed?

If no -- please squash migrations to have only one in PR

File: MangoAPI.Client/src/app/components/settings/settings.component.ts [hunk 4]
```diff
@@ -78,6 +79,12 @@ export class SettingsComponent implements OnInit, OnDestroy {
       .subscribe({
         next: (response) => {
           this.currentUser = response.user;
+          
+          console.log(response.user);
```
Comment (kolosovpetro) [inline MangoAPI.Client/src/app/components/settings/settings.component.ts:83]: remove this console log

File: MangoAPI.Client/src/app/components/settings/settings.component.ts [hunk 5]
```diff
@@ -147,6 +154,7 @@ export class SettingsComponent implements OnInit, OnDestroy {
       .pipe(takeUntil(this.componentDestroyed$))
       .subscribe({
         next: (response) => {
+          this.currentUser.userNameChanged = true;
           alert(response.message);
           this.ngOnInit();
```
Comment (kolosovpetro) [inline MangoAPI.Client/src/app/components/settings/settings.component.ts:159]: I think it is not necessary to call `ngOnInit` method here to update UI

**Review by kolosovpetro — CHANGES_REQUESTED** [review]

File: MangoAPI.Domain/Entities/UserEntity.cs (hunk 3 above)
Comment (Ketteiteki) [reply MangoAPI.Domain/Entities/UserEntity.cs:16]: Yes, I updated migrations

File: MangoAPI.BusinessLogic/ApiCommands/Users/UpdateUserAccountInfoCommandHandler.cs [hunk 6]
```diff
@@ -62,25 +62,14 @@ public class
...
-            }
-        }
-        
         user.UserInformation.BirthDay = request.BirthdayDate;
 
         user.UserInformation.Website = request.Website;
 
         user.UserName = request.Username;
 
+        user.UserNameChanged = true;
```
Comment (kolosovpetro) [inline MangoAPI.BusinessLogic/ApiCommands/Users/UpdateUserAccountInfoCommandHandler.cs:71]: What if user will pass there null or empty value?

File: MangoAPI.BusinessLogic/ApiCommands/Users/UpdateUserAccountInfoCommandValidator.cs [hunk 7]
```diff
@@ -22,6 +23,7 @@ public UpdateUserAccountInfoCommandValidator()
 
         RuleFor(x => x.Username)
             .Cascade(CascadeMode.Stop)
+            .Must(username => Guid.TryParse(username, out _) == false)
```
Comment (kolosovpetro) [inline MangoAPI.BusinessLogic/ApiCommands/Users/UpdateUserAccountInfoCommandValidator.cs:26]: in validator: permit only alpha numerical symbols

**Pull request merged**